packages/flutter_tools/lib/src/globals.dart

```diff
@@ -305,3 +306,5 @@ NonNullSafeBuilds get nonNullSafeBuilds => context.get<NonNullSafeBuilds>() ?? N
 /// A value of [null] indicates that no installation of java could be found on
 /// the host machine.
 Java? get java => context.get<Java>();
+
+TestCompilerNativeAssetsBuilder? get nativeAssetsBuilder => context.get<TestCompilerNativeAssetsBuilder>();
```

Comment:
I believe introducing new globals might be prohibited, I remember threading through the objects from main everywhere when introducing the indirection.

cc @christopherfujino

Reply:
I did it this way because I need to pass `nativeAssetsBuilder` to `FlutterDeviceManager` in `context_runner.dart` and wanted to keep all instantiations of `TestCompilerNativeAssetsBuilderImpl` in `exectubale.dart` and imports to `isolated` out of `context_runner.dart`. But that might not be the best approach. 😅

Reply:
> I believe introducing new globals might be prohibited

This was the truth at one point, but now it's complicated. Context: https://github.com/flutter/flutter/issues/47161. TL;DR I no longer prohibit globals when they make sense, but I would like to have globals dependency injected through a single DI object (similar to `BuildContext` in flutter widgets) rather than using zones. This is outside the scope of this PR.